viewer/src/utilities/RxCounter.ts

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { Subject, MonoTypeOperatorFunction } from 'rxjs';
+import { scan, tap, finalize, shareReplay } from 'rxjs/operators';
+
+enum Action {
+  Add,
+  Remove,
+  Clear
+}
+
+export class RxCounter {
+  private readonly _actionSubject: Subject<Action> = new Subject();
+
+  public add<T>(): MonoTypeOperatorFunction<T> {
+    return tap<T>({
+      next: () => this._actionSubject.next(Action.Add)
+    });
+  }
+
+  public remove<T>(): MonoTypeOperatorFunction<T> {
+    return tap<T>({
+      next: () => this._actionSubject.next(Action.Remove)
+    });
+  }
+
+  public clear<T>(): MonoTypeOperatorFunction<T> {
```

Comment:
Nitpick, but I'd call the functions `increment`, `decrement` and `reset` instead as add/remove/clear sound like a collection.

Reply:
I agree, I will do that